src/App.tsx

```diff
@@ -1,24 +1,244 @@
+  const [isEveryThingTrue, setIsEveryThingTrue] = useState(false);
+  const [toggleFalseTodosId, setToggleFalseTodosId] = useState([0]);
+  const [todoStatusChange, setTodoStatusChange] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const excludedInputRef = useRef<HTMLInputElement>(null);
+  const timeoutId = useRef<NodeJS.Timeout | null>(null);
+
+  const fetchTodos = async () => {
+    try {
+      setIsLoading(true);
+      const response = await getTodos(USER_ID);
+
+      setTodos(response);
+    } catch (error) {
+      setIsHidden('Unable to add a todo');
+      setIsThereIssue(true);
+      timeoutId.current = setTimeout(() => {
+        setIsThereIssue(false);
+      }, 3000);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const updateTempTodo = (value: string) => {
+    setInputValue(value);
+  };
+
+  const handleFormSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsThereIssue(false);
+    setIsLoading(true);
+    setIsPlusOne(true);
+
+    setTempTodo({
+      ...tempTodo,
+      title: inputValue.trim(),
+      id: 0,
+      completed: false,
+      userId: USER_ID,
+    });
+
+    if (inputValue.trim() !== '') {
+      const tempTodoItem: Todo = {
+        title: inputValue.trim(),
+        userId: USER_ID,
+        completed: false,
+        id: +nanoid(),
+      };
+
+      try {
+        await client.post('/todos', tempTodoItem);
+        setTodos((prevTodo) => [...prevTodo, tempTodoItem]);
+        setApiResponseReceived(false);
+      } catch (error) {
+        setIsThereIssue(true);
+        setDeleteErrorMessage('Unable to add the todo');
+        timeoutId.current = setTimeout(() => {
+          setIsThereIssue(false);
+        }, 3000);
+        setIsPlusOne(false);
+      } finally {
+        setApiResponseReceived(true);
+        setIsLoading(false);
+        setIsPlusOne(false);
+      }
+    } else {
+      setIsThereIssue(true);
+      setIsPlusOne(false);
+      setIsTitleEmpty('Title can\'t be empty');
+    }
+
+    setInputValue('');
+  };
+
+  const deleteTodo = async (id: number) => {
+    setDeletedTodoId(id);
+
+    try {
+      setIsLoading(true);
+      await client.delete(`/todos/${id}`);
+      setTodos(todos.filter((todo) => todo.id !== id));
+    } catch (error) {
+      setDeleteErrorMessage('Unable to delete the todo');
+      setIsThereIssue(true);
+      timeoutId.current = setTimeout(() => {
+        setIsThereIssue(false);
+      }, 3000);
+      setIsLoading(false);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchTodos();
+
+    return () => {
+      if (timeoutId.current) {
+        clearTimeout(timeoutId.current);
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    const isActive = todos.some((obj) => obj.completed === false);
+    const isFalse = todos.some((obj) => obj.completed === true);
+    const todoLength = todos.filter((obj) => {
```

Comment:
Don't use useEffect() for it. You can use [useMemo](https://react.dev/reference/react/useMemo)() for the variables.